internal/lib/stats/cpu_metrics.go

```diff
@@ -0,0 +1,105 @@
+package statsserver
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cri-o/cri-o/internal/config/cgmgr"
+	"github.com/cri-o/cri-o/internal/lib/sandbox"
+	types "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+func generateSandboxCPUMetrics(sb *sandbox.Sandbox, cpu *cgmgr.CPUStats) []*types.Metric {
+	cpuMetrics := []*containerMetric{
+		{
+			desc: &types.MetricDescriptor{
+				Name:      "container_cpu_user_seconds_total",
+				Help:      "Cumulative user CPU time consumed in seconds.",
+				LabelKeys: baseLabelKeys,
+			},
+			valueFunc: func() metricValues {
+				return metricValues{{
+					value:      cpu.UsageInUsermode / uint64(time.Second),
+					metricType: types.MetricType_COUNTER,
+				}}
+			},
+		}, {
+			desc: &types.MetricDescriptor{
+				Name:      "container_cpu_system_seconds_total",
+				Help:      "Cumulative system CPU time consumed in seconds.",
+				LabelKeys: baseLabelKeys,
+			},
+			valueFunc: func() metricValues {
+				return metricValues{{
+					value:      cpu.UsageInKernelmode / uint64(time.Second),
+					metricType: types.MetricType_COUNTER,
+				}}
+			},
+		}, {
+			desc: &types.MetricDescriptor{
+				Name:      "container_cpu_usage_seconds_total",
+				Help:      "Cumulative CPU time consumed in seconds.",
+				LabelKeys: append(baseLabelKeys, "cpu"),
+			},
+			valueFunc: func() metricValues {
+				if len(cpu.PerCPUUsage) == 0 {
+					if cpu.TotalUsageNano > 0 {
+						return metricValues{{
+							value:      cpu.TotalUsageNano / uint64(time.Second),
+							labels:     []string{"total"},
+							metricType: types.MetricType_COUNTER,
+						}}
+					}
+				}
```

Comment:
Collapse to create an association for both statements to reduce indentation, perhaps?

```suggestion
				if len(cpu.PerCPUUsage) == 0 && cpu.TotalUsageNano > 0 {
					return metricValues{{
						value:      cpu.TotalUsageNano / uint64(time.Second),
						labels:     []string{"total"},
						metricType: types.MetricType_COUNTER,
					}}
				}
```

Reply:
done